### PR TITLE
[RPC] accelerate sampling RPC

### PR DESCRIPTION
**Pull request by VoVAllen**
## Description
Fix send None, Null NDArray and accelerate sampling RPC

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by zheng-da** [comment]
why do you merge with this PR? https://github.com/dmlc/dgl/pull/1653

It's better to not put everything in the same PR.

File: python/dgl/distributed/rpc.py [hunk 1]
```diff
@@ -419,14 +424,14 @@ def deserialize_from_payload(cls, data, tensors):
         De-serialized object of class cls.
     """
     pos, nonarray_state = pickle.loads(data)
-    state = [None] * (len(nonarray_state) + len(tensors))
+    state = [PLACEHOLDER] * (len(nonarray_state) + len(tensors))
```
Comment (zheng-da) [inline python/dgl/distributed/rpc.py:427]: i think you should add some comments to explain why you want to use `PLACEHOLDER`

File: python/dgl/distributed/sampling.py [hunk 2]
```diff
@@ -83,17 +83,15 @@ def sample_neighbors(dist_graph, nodes, fanout, edge_dir='in', prob=None, replac
     assert edge_dir == 'in'
     req_list = []
     partition_book = dist_graph.get_partition_book()
-
+    np_nodes = np.array(nodes)
```
Comment (zheng-da) [inline python/dgl/distributed/sampling.py:86]: what happens if `nodes` is a DL tensor?

Comment (VoVAllen) [reply python/dgl/distributed/sampling.py]: Fixed by using Index

**Pull request merged**